plugins/analysis-icu/src/main/java/org/elasticsearch/index/mapper/ICUCollationKeywordFieldMapper.java

```diff
@@ -642,6 +659,10 @@ protected void doMerge(Mapper mergeWith) {
             conflicts.add("Cannot update hiragana_quaternary_mode setting for [" + CONTENT_TYPE + "]");
         }
 
+        if (ignoreAbove != icuMergeWith.ignoreAbove) {
```

Comment:
I think we should actually allow `ignore_above` to be updated during a merge (see https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/mapper/KeywordFieldMapper.java#L390-L394 for what we do for `keyword` fields).